### ISSUE TITLE
[MPIR-221] Fix for MPIR-171 generates unnecessary and confusing warnings for non-integral time-zones

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-project-info-reports-plugin#159]
**[Sebb](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=sebb@apache.org)** opened **[MPIR-221](https://issues.apache.org/jira/browse/MPIR-221?redirect=false)** and commented

The change for MPIR-171 added a check to see if the timezone is numeric. If not, it reports a warning.

However, the check fails to allow for non-integral timezones - e.g. 5.5 - and the warning message is incorrect, as it does not show the correct information.

For example, the following setting:

    <developer>
      <timezone>+5.5</timezone>
    </developer>

results in the warning:

[WARNING] The time zone '-2147483648' for the developer 'Name' is not a recognised time zone, use a number in the range -12 to +14 instead of.

The message should read something like:

[WARNING] The time zone '+5.5' for the developer 'Name' is not a recognised time zone, use a number in the range -12 to +14 or a valid long TimeZone id.

except that it should not be generated for that particular timezone, as +5.5 is as valid as +5 (which does not generate a warning).

To summarise:
1) the warning message is not helpful, as it does not show the correct timezone.
2) the warning should not be generated for fractional timezones.

---

**Affects:** 2.3.1
